moquette/broker/src/main/java/io/moquette/broker/Session.java

```diff
@@ -113,6 +113,34 @@ public int compareTo(Delayed o) {
         this.sessionQueue = sessionQueue;
     }
 
+    /**
+     * This method terminates a Session, which involves disconnecting the client,
+     * if one is connected, and releasing all references to unacknowledged publishes.
+     */
+    void terminateSession() {
+        MQTTConnection connection = this.mqttConnection;
+        if (connection != null) {
+            connection.dropConnection();
+        }
+        disconnect();
```

Comment:
what does `disconnect` do if connection is null?

for that matter, what does `dropConnection` do that wouldn't be done in `disconnect`?

I'm just confused about why you wouldn't do connection management in `disconnect`

Reply:
`disconnect` is a misnomer. It doesn't actually touch the TCP connection. It just modifies internal Session state. `dropConnection` drops the actual connection.

But now that I look at this again, `dropConnection` can result in (some, but not all scenarios) this method getting called again. Will re-visit this. It's a bit messy